firefox-ios/Client/Frontend/UserContent/UserScripts/AddressFormManager/AddressFormManager.mjs

```diff
@@ -5,19 +5,58 @@
 import "resource://gre/modules/shared/Helpers.ios.mjs";
 import { createFormLayoutFromRecord } from "resource://gre/modules/shared/addressFormLayout.mjs";
 
+/**
+ * Sets the theme of the webview.
+ * @param {Boolean} isDarkTheme - Set to true if the dark theme should be applied.
+ */
+const setTheme = (isDarkTheme) => {
+  document.body.classList.toggle("dark", isDarkTheme);
+};
+
+/**
+ * Automatically resizes a textarea to fit its content.
+ * @param {HTMLTextAreaElement} textarea - The textarea element to resize.
+ * @param {Number} initialHeight - The initial height of the textarea. Default is 22.
+ */
+const autoResizeTextarea = (textarea, initialHeight = 22) => {
+  textarea.style.height = "auto";
+  const lines = textarea.value.match(/^/gm).length;
+
+  if (lines <= 1) {
+    textarea.style.height = `${initialHeight}px`;
+  } else {
+    textarea.style.height = `${textarea.scrollHeight}px`;
+  }
+};
+
 /**
  * Renders an address form into the webview from a record.
  *
  * @param {object} record - Address record, includes at least country code defaulted to FormAutofill.DEFAULT_REGION.
  * @param {object} l10nStrings - Localization strings map.
  */
-const init = (record, l10nStrings) => {
+const init = (record, l10nStrings, isDarkTheme = false) => {
+  setTheme(isDarkTheme);
+
+  // Replace all "\\n" with new line character.
+  // TODO: We should probably do this Swift code instead.
+  for (const [key, value] of Object.entries(record)) {
+    if (typeof value === "string") {
+      record[key] = value.replaceAll("\\n", "\n");
+    }
+  }
+
```

Comment:
@razvanlitianu I am doing this in JS since the strings I get from swift have new lines eascaped `\\n` instead of `\n`. I think this is what's causing it:

https://github.com/mozilla-mobile/firefox-ios/blob/52ab80f6cd5830452f7171ea6e3b469468a9f598/firefox-ios/Client/Frontend/Autofill/Address/AddressListViewModel.swift#L137

I am fine with doing here in JS. I was just wondering if it's easier to do it  in swift instead ?

Reply:
Doing it in js is fine for me, we can remove the swift code

Reply:
Cool. `TODO` removed in https://github.com/mozilla-mobile/firefox-ios/pull/20427/commits/72d86f643265b569ea38ba27e5180734a4218f97.